pyi.py

```diff
@@ -1978,8 +1978,8 @@ def run(self) -> Iterable[Error]:
             tree = LegacyNormalizer().visit(self.tree)
             yield from PyiVisitor(filename=path).run(tree)
 
-    @classmethod
-    def add_options(cls, parser: OptionManager) -> None:
+    @staticmethod
+    def add_options(parser: OptionManager) -> None:
```

Comment:
Once we drop support for Python 3.7, we can drop support for running with flake8 v5, which should mean we'll be able to get rid of the `optparse`-related code in this method. flake8 v6 no longer supports `optparse`.